src/commonmark.c

```diff
@@ -477,7 +477,13 @@ static int S_render_node(cmark_renderer *renderer, cmark_node *node,
   case CMARK_NODE_FOOTNOTE_REFERENCE:
     if (entering) {
       LIT("[^");
-      OUT(cmark_chunk_to_cstr(renderer->mem, &node->as.literal), false, LITERAL);
+
+      char *footnote_label = renderer->mem->calloc(1, (sizeof(char) * node->parent_footnote_def->as.literal.len) + 1);
```

Comment:
If we have a macro that checks for `size_t` overflow, we'd want to use it here, because there is the possibility of an overflow here, especially on a 32-bit system.

Because of the definition of `calloc`, we'd probably want to write this:

```
char *footnote_label = renderer->mem->calloc(node->parent_footnote_def->as.literal.len + 1, sizeof(char));
```

As a note, `sizeof(char)` is defined as 1 in the C standard, but writing it this way is better than making that assumption.

Reply:
:+1: